### PR TITLE
Feature/Add visualizations to Indicators and Program Indicators

### PR DESCRIPTION
**Pull request by nshandra**
### References
Issues:
-  [Add information about Indicators](https://app.clickup.com/t/3jna4bf)
-  [Add information about Program Indicators](https://app.clickup.com/t/3jna3zr)

### Description
- Adapt the indicator visualization from the "Search" tab for Indicators and Program Indicators.
- The Indicators visualization displays the Indicator and the dataElements and Program Indicators present in its numerator or denominator.
- The Program Indicators visualization only displays the Program Indicator.

### CODE REVIEW

File: app/dossierPrograms/dossierPrograms.controllers.js [hunk 1]
```diff
@@ -206,6 +206,202 @@ dossierProgramsModule.controller("dossiersProgramSectionController", [
...
+            payload.name = name + " - " + id;
+            payload.rows[0].items = rowItems ? items.concat(rowItems.map(id => ({ id: id }))) : items;
+
+            $scope.table = dossiersProgramVisualizationTableFactory.get_table.query(
+                {
+                    filter: "name:eq:" + payload.name,
+                },
+                function (tbl) {
+                    if (tbl && tbl.visualizations[0]) {
+                        if (tbl.visualizations[0].id) {
```
Comment (tokland) [inline app/dossierPrograms/dossierPrograms.controllers.js]: [minor] `tbl.visualizations[0].id` used 6 times, so `const visualizationId = tbl.visualizations[0].id`.

File: app/dossierPrograms/dossierPrograms.controllers.js [hunk 2]
```diff
@@ -206,6 +206,202 @@ dossierProgramsModule.controller("dossiersProgramSectionController", [
...
+
+                                    const uid = tbl.visualizations[0].id;
+
+                                    $window.open(dhisroot + "/dhis-web-data-visualizer/index.html#/" + uid, "_blank");
+                                }
+                            );
+                        }
+                    }
+
+                    if (tbl.visualizations[0] == undefined) {
```
Comment (tokland) [inline app/dossierPrograms/dossierPrograms.controllers.js]: [minor] is this an `else` of the previous branch?

File: app/dossierPrograms/dossierPrograms.controllers.js [hunk 3]
```diff
@@ -206,6 +206,202 @@ dossierProgramsModule.controller("dossiersProgramSectionController", [
...
+                    publicAccess: "--------",
+                    externalAccess: false,
+                    userGroupAccesses: [{ id: "epFY01iJN0Z", name: "ALL USERS", access: "rw------" }],
+                },
+            };
+
+            let items = [{ id: id }];
+
+            payload.name = name + " - " + id;
+            payload.rows[0].items = rowItems ? items.concat(rowItems.map(id => ({ id: id }))) : items;
```
Comment (tokland) [inline app/dossierPrograms/dossierPrograms.controllers.js:353]: [minor, no need to change] I guess we want a base payload which can be easily copy/pasted, that's why we modify it here. With a functional approach, we would have a `basePayload` and then `const payload = {...basePayload, ...}` with all the required changes.

**Review by tokland — COMMENTED** [review]
[code-only review] Minor comments inline:

File: app/dossierPrograms/dossierPrograms.controllers.js (hunk 1 above)
Comment (nshandra) [reply app/dossierPrograms/dossierPrograms.controllers.js]: Done, it should make thing more readable.

File: app/dossierPrograms/dossierPrograms.controllers.js (hunk 2 above)
Comment (nshandra) [reply app/dossierPrograms/dossierPrograms.controllers.js]: Yes, I overlooked that, done now.

File: app/dossierPrograms/dossierPrograms.controllers.js (hunk 3 above)
Comment (nshandra) [reply app/dossierPrograms/dossierPrograms.controllers.js:353]: i reused the code from `app/search/search.controllers.js`  so i kept most of it 'as is', but you are obviously right.

**Review by tokland — APPROVED** [review]

**Pull request merged**